File: sesion-5/secure-snippets-js-csharp/csharp/2-Secrets.Fixed.cs
// ✅ Mitigación: Secretos desde variables de entorno o secretos de usuario
using System;

class SecretsFixed {
    static void Main() {
        var apiKey = Environment.GetEnvironmentVariable("API_KEY");
        var dbPassword = Environment.GetEnvironmentVariable("DB_PASSWORD");
        if (string.IsNullOrEmpty(apiKey) || string.IsNullOrEmpty(dbPassword)) {
            Console.Error.WriteLine("Faltan secretos. Configure API_KEY y DB_PASSWORD.");
            Environment.Exit(1);
        }
        Console.WriteLine("API Key cargada (****)");
        // Usar dbPassword sin imprimirla nunca
    }
}

File: sesion-5/secure-snippets-js-csharp/js/2-secrets.fixed.js
/**
 * ✅ Mitigación: Secretos vía variables de entorno (.env) o secret managers
 * - Nunca hardcodear API keys.
 * - Falla rápido si falta un secreto.
 */
require('dotenv').config(); // opcional si usas .env
const API_KEY = process.env.API_KEY;
const DB_PASSWORD = process.env.DB_PASSWORD;
if (!API_KEY || !DB_PASSWORD) {
  throw new Error('Faltan secretos: defina API_KEY y DB_PASSWORD en variables de entorno');
}

function connect() {
  console.log("Conectando con API_KEY (masked):", API_KEY.substring(0,4) + '...');
  // Conectar usando DB_PASSWORD sin loguearla
}
connect();
